ARCHITECTURE.md

```diff
@@ -41,4 +42,16 @@ For more details about the models we use in Scanner see web-monitoring-db's [API
+|B| [Internet Archive's Wayback Machine is queried for imports](https://archive.readme.io/docs) | [web-monitoring-processing](https://github.com/edgi-govdata-archiving/web-monitoring-processing) - this [PR](https://github.com/edgi-govdata-archiving/web-monitoring-processing/pull/174) | Data is pulled from the IA API, formatted, and [POST](https://api.monitoring.envirodatagov.org/#/imports/post_imports_)ed to [-db](https://github.com/edgi-govdata-archiving/web-monitoring-db) | Runs on a cron job, formats per [GH issue comment](https://github.com/qri-io/walk/issues/16#issuecomment-437785099) |
+|C| New metadata arrives at [-db](https://github.com/edgi-govdata-archiving/web-monitoring-db) and is stored in a database | Scripts running on cron jobs that do ETL via scrapes or APIs (These scripts: [Versionista scraper](https://github.com/edgi-govdata-archiving/web-monitoring-versionista-scraper), [Wayback Machine importer](https://github.com/edgi-govdata-archiving/web-monitoring-processing))| [POST](https://api.monitoring.envirodatagov.org/#/imports/post_imports_) to [/api/v0/imports](https://api.monitoring.envirodatagov.org/#/imports/post_imports_) as JSON array or newline-delimited JSON stream (stream preferred) | Contains:<ul><li> a URL for where to retrieve the raw response body of the version</li><li> a SHA-256 hash of that data</li></ul> Example data in [GH issue comment](https://github.com/qri-io/walk/issues/16#issuecomment-437785099)|
+|D| Determination is made whether or not to download the data | [-db](https://github.com/edgi-govdata-archiving/web-monitoring-db) | If the URL is in an acceptable, publicly readable location |  Acceptable location checked [per .env](https://github.com/edgi-govdata-archiving/web-monitoring-db/blob/30d976676935f5f621e18285016a137c8b55a0d8/.env.example#L26-L27) |
+|    ↳ E1| URL for raw data is stored  | [-db](https://github.com/edgi-govdata-archiving/web-monitoring-db) (URI in metadata is untouched)| [-db](https://github.com/edgi-govdata-archiving/web-monitoring-db) stores URL | Happens if [-db](https://github.com/edgi-govdata-archiving/web-monitoring-db) has determined not to download the data |
+|    ↳ E2| Raw data is verified and stored on URL we store and maintain | [-db](https://github.com/edgi-govdata-archiving/web-monitoring-db) (URI in metadata is changed to point to our new location) | Raw response data downloaded from the URL, verified against SHA-256 hash of the data from initial POST, and stores it (in production) in a public S3 bucket | Happens if [-db](https://github.com/edgi-govdata-archiving/web-monitoring-db) has determined to download the data |
+| F | (Success state, data has been stored) | | | |
+
+[![diagram render](https://i.imgur.com/T1Ej92x.png)](https://hackmd.io/-EAU-6WpQ0ypWzZ1ytyztA)
```

Comment:
I’m cool if you want to leave these imgur and hackmd link, but I was just thinking you’d put both the image and the source file in this repo, so you don’t have to be online to read this part of the doc, and so that it all stays managed together, with no worries about synchronizing changes in content or content disappearing in one place and not another.

Reply:
I see your point, but think I'll leave the hackmd in place since it (unlike github) renders mermaid & so it's a convenient way to keep the source & re-render at the same time. Will replace the image link with a local one though.